packages/pharos/src/components/coach-mark/pharos-coach-mark.ts

```diff
@@ -0,0 +1,165 @@
+  static elementDefinitions = {
+    'pharos-button': PharosButton,
+    'pharos-heading': PharosHeading,
+  };
+
+  private offsetX = 0;
+  private offsetY = 0;
+  private margin = 20;
+
+  public static override get styles(): CSSResultArray {
+    return [coachMarkStyles];
+  }
+
+  /**
+   * Indicates that the coach mark should not be displayed on the page
+   * @attr hide
+   */
+  @property({ type: Boolean, reflect: true })
+  public hide = true;
+
+  /**
+   * Indicates which side of the base element the coachmark should appear on
+   * @attr side
+   * @type {Side}
+   */
+  @property({ reflect: true })
+  public side: Side = 'bottom';
+
+  /**
+   * Indicates how the coach mark carat should be aligned in relation to the coach mark content
+   * @attr alignment
+   * @type {Alignment}
+   */
+  @property({ reflect: true })
+  public alignment: Alignment = 'center';
+
+  /**
+   * Dictates how long to wait between coach mark trigger and start of coach mark fade in animation
+   * @attr delay
+   * @type {Delay}
+   */
+  @property({ reflect: true })
+  public delay: Delay = 'short';
+
+  /**
+   * Text content for the coach mark header
+   * @attr header
+   * @type {String}
+   */
+  @property({ reflect: true })
+  public header = '';
+
+  /**
+   * Style variant
+   * @attr variant
+   * @type {Variant}
+   */
+  @property({ reflect: true })
+  public variant: Variant = 'light';
+
+  /**
+   * Set minimum width of coach mark
+   * @attr width
+   * @type {number}
+   */
+  @property({ reflect: true })
+  public width = 250;
+
+  constructor() {
+    super();
+    this.setOffset();
+    this.resizeObserver.observe(document.body);
+  }
+
+  private resizeObserver = new ResizeObserver(
+    debounce((entries) => {
+      entries.map(() => {
+        this.setOffset();
+        this.requestUpdate();
+      });
+    }, 100)
+  );
+
+  private setOffset() {
+    const id: string = this.getAttribute('id') || '';
+    const targetElement: Element | null = document.querySelector(`[data-coach-mark="${id}"]`);
+    if (!targetElement) return;
+    const thisRect = this.getBoundingClientRect();
+    const targetRect = targetElement.getBoundingClientRect();
+
+    switch (this.side) {
+      case 'top':
+        this.offsetX = (targetRect.left + targetRect.right) / 2 - thisRect.x;
+        this.offsetY = targetRect.top - thisRect.bottom - this.margin;
+        break;
+      case 'right':
+        this.offsetX = targetRect.right - thisRect.left + this.margin;
+        this.offsetY = (targetRect.top + targetRect.bottom) / 2 - thisRect.y;
+        break;
+      case 'left':
+        this.offsetY = (targetRect.top + targetRect.bottom) / 2 - thisRect.y;
+        this.offsetX = targetRect.left - thisRect.right - this.margin;
+        break;
+      default: // bottom
+        this.offsetX = (targetRect.left + targetRect.right) / 2 - thisRect.x;
+        this.offsetY = targetRect.bottom - thisRect.top + this.margin;
+    }
```

Comment:
In Tooltip we leverage `floating-ui` to do a lot of the math for us; are we opening ourselves to maintenance burden and human error here?

Reply:
Good call out. By using `floating-ui` I was able to remove the calculations and reduce the amount of styling required for element positioning